Core/OfficeDevPnP.Core/Resources/SP-Responsive-UI.js

```diff
@@ -1,117 +1,121 @@
-            }
-
-            PnPResponsiveApp.setUpToggling();
-			PnPResponsiveApp.responsivizeSettings();
-			
-			// also listen for dynamic page change to Settings page
-			window.onhashchange = function () { PnPResponsiveApp.responsivizeSettings(); };
-			
-			// extend/override some SP native functions to fix resizing quirks
-			var originalResizeFunction = FixRibbonAndWorkspaceDimensions;
-			FixRibbonAndWorkspaceDimensions = function() {
-				// let sharepoint do its thing
-				originalResizeFunction();
-				// fix the body container width
-				$("#s4-bodyContainer").width($("#s4-workspace").width() );
-			}
+    PnPResponsiveApp.responsivizeSettings = function () {
+        // return if no longer on Settings page
+        if (window.location.href.indexOf('/settings.aspx') < 0) return;
+
+        // find the Settings root element, or wait if not available yet
+        var settingsRoot = $(".ms-siteSettings-root");
+        if (!settingsRoot.length) {
+            setTimeout(PnPResponsiveApp.responsivizeSettings, 100);
+            return;
+        }
+
+        $(".ms-siteSettings-root .ms-linksection-level1").each(function () {
+            var self = $(this),
+                settingsDiv = $('<div>');
+            settingsDiv.addClass("pnp-settingsdiv");
+            self.find(".ms-linksection-iconCell img").appendTo(settingsDiv);
+            self.find(".ms-linksection-textCell").children().appendTo(settingsDiv);
+            settingsDiv.appendTo(settingsRoot);
         });
+        settingsRoot.find("table").remove();
+    };
+
+
+    PnPResponsiveApp.setUpToggling = function () {
+        // if it is already responsivized, return
+        if ($("#navbar-toggle").length)
+            return;
+
+        // Set up sidenav toggling
+        var topNav = $('#DeltaTopNavigation'),
+            topNavClone = topNav.clone();
+        topNavClone.addClass('mobile-only');
+        topNavClone.attr('id', topNavClone.attr('id') + "_mobileClone");
+        topNav.addClass('no-mobile');
+        $('#sideNavBox').append(topNavClone);
+
+        var sideNavToggle = $('<button id="navbar-toggle" class="mobile-only burger" type="button"><span></span></button>');
+        sideNavToggle.click(function (event) {
+            event.preventDefault();
+            $("body").toggleClass('shownav');
+            $(this).toggleClass('selected');
+        });
+        $("#pageTitle").before(sideNavToggle);
+    };
+
+    PnPResponsiveApp.init = function () {
+        if (!window.jQuery) {
+            // jQuery is needed for PnP Responsive UI to run, and is not fully loaded yet, try later
+            setTimeout(PnPResponsiveApp.init, 100);
+        } else {
+            $(function () { // only execute when DOM is fully loaded
+
+                // embedding and loading of all necessary CSS files and JS libraries
+                var currentScriptUrl = $('#PnPResponsiveUI').attr('src');
+                if (currentScriptUrl != "undefined") {
```

Comment:
Hi Edwin,
There is still one issue in this line. Could you please fix it before we can evaluate and eventually merge this PR?
Thanks.